rules_python/whl.py

```diff
@@ -58,7 +58,7 @@ def metadata(self):
     # directory.
     with zipfile.ZipFile(self.path(), 'r') as whl:
       with whl.open(os.path.join(self._dist_info(), 'metadata.json')) as f:
-        return json.loads(f.read())
+        return json.loads(f.read().decode("utf-8"))
```

Comment:
Looking at the `json.loads` function, it seems like this should (originally) have just been `return json.load(f)` (note change from `loads` to `load`), which handles all the encoding/decoding/python2-vs-3.  Can you try that and see if it does indeed work for you?

Reply:
Tried it, and it actually didn't work. When I called load instead of loads, I got the following traceback:
```bash
Traceback (most recent call last):
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 131, in <module>
    main()
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 127, in main
    for d in whl.dependencies()
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 126, in <listcomp>
    'requirement("%s")' % d
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 68, in dependencies
    run_requires = self.metadata().get('run_requires', [])
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 61, in metadata
    return json.load(f.read())
  File "/usr/lib/python3.5/json/__init__.py", line 265, in load
    return loads(fp.read(),
AttributeError: 'bytes' object has no attribute 'read'
```

Reply:
Oh oops, I was passing "f.read()" to the load function instead of passing "f". I tried again, this time correctly, and I get the same str/bytes related error, but in a deeper function call. Here's the traceback:
```bash
Traceback (most recent call last):
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 131, in <module>
    main()
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 127, in main
    for d in whl.dependencies()
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 126, in <listcomp>
    'requirement("%s")' % d
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 68, in dependencies
    run_requires = self.metadata().get('run_requires', [])
  File "/home/kevin/.cache/bazel/_bazel_kevin/cb16c580e8d350a5b65c591962f69d45/external/io_bazel_rules_python/rules_python/whl.py", line 61, in metadata
    return json.load(f)
  File "/usr/lib/python3.5/json/__init__.py", line 268, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

Reply:
Ok, then what you originally have is fine.

Can you tell me what operating system and Python version you're using?  There's a variety of important differences between 2.6, 2.7, and 3.x that are relevant to this bit of code.

We don't have good test coverage for those differences either, which is a separate issue.

Reply:
I am on Ubuntu 16.04.

I tested this with python 3.5.2 and 2.7.12.

To test, I used a virtualenv with python3 installed. Built my sample repo using bazel, and then ran my binary. Then I ran "bazel clean --expunge", opened a new terminal without the virtualenv activated (i.e. python 2.7), rebuilt with bazel, and re-ran. Both worked for me.

I also confirmed that without this patch, the bazel build step fails with python3, but works with python2.7.